climada_petals/util/cds_glofas_downloader.py

```diff
@@ -0,0 +1,256 @@
+from pathlib import Path
+import multiprocessing as mp
+from copy import deepcopy
+from typing import Iterable, Mapping, Any, Optional, List, Union
+from itertools import repeat
+from datetime import date, timedelta
+import logging
+
+from cdsapi import Client
+from ruamel.yaml import YAML
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_REQUESTS = {
+    "historical": {
+        "variable": "river_discharge_in_the_last_24_hours",
+        "product_type": "consolidated",
+        "system_version": "version_3_1",
+        "hydrological_model": "lisflood",
+        "format": "grib",
+        "hyear": "1979",
+        "hmonth": [
+            "january",
+            "february",
+            "march",
+            "april",
+            "may",
+            "june",
+            "july",
+            "august",
+            "september",
+            "october",
+            "november",
+            "december",
+        ],
+        "hday": [f"{day:02}" for day in range(1, 32)],
+    },
+    "forecast": {
+        "variable": "river_discharge_in_the_last_24_hours",
+        "product_type": "ensemble_perturbed_forecasts",
+        "system_version": "operational",
+        "hydrological_model": "lisflood",
+        "format": "netcdf",
+        "year": "2022",
+        "month": "08",
+        "day": "01",
+        "leadtime_hour": list(map(str, range(1, 31))),
+    },
+}
+"""Default request keyword arguments to be updated by the user requests"""
```

Comment:
I don't understand why you would have a `util` module for this functionality in the first place. The functions in `cds_glofas_downloader` are only used in the `climada_petals.hazard.rf_glofas` subpackage, so why not create it as a new module in that subpackage? The whole module seems very application-specific to me, it even has `glofas` in its name. I think it's completely safe to move it to `climada_petals.hazard.rf_glofas.cds_downloader`.

Reply:
True. I envisaged that this module can be easily extended to download other data than those of GloFAS from the CDS, hence I put it there. I'll move it.

Reply:
I moved the download module into the hazard submodule.